src/libraries/System.Drawing.Common/src/System/Drawing/DrawingComWrappers.cs

```diff
@@ -0,0 +1,331 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Drawing
+{
+    /// <summary>
+    /// The ComWrappers implementation for System.Drawing.Common's COM interop usages.
+    ///
+    /// Supports IStream and IPicture COM interfaces.
+    /// </summary>
+    internal unsafe class DrawingComWrappers : ComWrappers
+    {
+        private const int OK = 0;
+        private static readonly Guid IStreamIID = new Guid(0x0000000C, 0x0000, 0x0000, 0xC0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46);
+
+        private static readonly ComInterfaceEntry* s_wrapperEntry = InitializeComInterfaceEntry();
+
+        internal static DrawingComWrappers Instance { get; } = new DrawingComWrappers();
+
+        private DrawingComWrappers() { }
+
+        internal static void CheckStatus(int result)
+        {
+            if (result != OK)
```

Comment:
The usual HRESULT failure check is `result < 0`. It is what `ThrowExceptionForHR` does, so you can just use that instead of CheckStatus. 

Or is there a particular reason why you are using `result == 0`?

Reply:
No particular reason. I didn't know about `Marshal.ThrowExceptionForHR`. Converting to that instead.